src/System.Net.Mail/src/System/Net/Mail/SmtpClient.cs

```diff
@@ -136,21 +136,7 @@ private void Initialize()
                 // machine connecting to public server).
 
                 // SMTP RFC's require ASCII only host names in the HELO/EHLO message.
-
-                // TODO: https://github.com/dotnet/corefx/issues/19605
-                //
-                //   Catching NotSupportedException here because HostName is currently PNSE
-                //   on UapAot. HostName is planned to be restored for .NetCore 2.1
-                //   so once that's done, the try-catch can be removed.
-                string clientDomainRaw;
-                try
-                {
-                    clientDomainRaw = IPGlobalProperties.GetIPGlobalProperties().HostName;
-                }
-                catch (NotSupportedException)
-                {
-                    clientDomainRaw = "LocalHost";
-                }
+                string clientDomainRaw = IPGlobalProperties.GetIPGlobalProperties().HostName;
```

Comment:
> A platform policy is that synchronous I/O (or anything blocking) is prohibited on the UI thread of an UWP application (it will crash the app).

I'm not convinced this is true. While an app will be terminated if it takes too long to launch/resume, the same hasn't been true for blocking the UI thread. @CIPop were you thinking of the launch/resume case?